Source/artifacts/ArtifactsManager.js

```diff
@@ -32,95 +32,129 @@ export class ArtifactsManager {
             .then(context => {
                 this._logger.info(`Creating read model with name '${context.name}' and namespace '${context.namespace}'`);
-                _boilerPlatesManager.get(this).createArtifactInstance('readModel', language, boilerPlate, destination, context);  
+                _boilerPlatesManager.get(this).createArtifactInstance('readModel', flags.language, boilerPlate, destination, context);  
             });
     }
     /**
      * Create an aggregate root
-     * @param {string} name 
-     * @param {string} namespace 
+     * @param {any} flags
      */
-    createAggregateRoot(language) {
+    createAggregateRoot(flags) {
 
-        let boilerPlate = _boilerPlatesManager.get(this).boilerPlatesByType('artifacts')[0];
+        let boundedContextConfig = this._getNearestBoundedContextConfig();
+        flags.language = boundedContextConfig.language;
+        let boilerPlate = this._getArtifactBoilerPlateByLanguageAndType(flags.language, 'command');
         let destination = process.cwd();
-        _inquirerManager.get(this).promptForAggregateRoot(language)
+
+        _inquirerManager.get(this).promptForAggregateRoot(flags)
             .then(context => {
                 this._logger.info(`Creating aggregate root with name '${context.name}' and namespace '${context.namespace}'`);
-                _boilerPlatesManager.get(this).createArtifactInstance('aggregateRoot', language, boilerPlate, destination, context);  
+                _boilerPlatesManager.get(this).createArtifactInstance('aggregateRoot', flags.language, boilerPlate, destination, context);  
             });
     }
     /**
      * Create a query
-     * @param {string} language
+     * @param {any} flags
      */
-    createQuery(language) {
+    createQuery(flags) {
 
-        let boilerPlate = _boilerPlatesManager.get(this).boilerPlatesByType('artifacts')[0];
+        let boundedContextConfig = this._getNearestBoundedContextConfig();
+        flags.language = boundedContextConfig.language;
+        let boilerPlate = this._getArtifactBoilerPlateByLanguageAndType(flags.language, 'command');
         let destination = process.cwd();
 
-        _inquirerManager.get(this).promptForQuery(language)
+        _inquirerManager.get(this).promptForQuery(flags)
             .then(context => {
                 this._logger.info(`Creating query with name '${context.name}'`)
-                _boilerPlatesManager.get(this).createArtifactInstance('query', language, boilerPlate, destination, context);  
+                _boilerPlatesManager.get(this).createArtifactInstance('query', flags.language, boilerPlate, destination, context);  
             });
     }
     /**
      * Create a query for a specific read model
-     * @param {string} language
+     * @param {any} flags
      */
-    createQueryFor(language) {
-        
-        let boilerPlate = _boilerPlatesManager.get(this).boilerPlatesByType('artifacts')[0];
+    createQueryFor(flags) {
```

Comment:
Terminology; flags is typical bits / booleans one turns on and off. I'd switch to something like context.